pkg/infrastructure/aws/clusterapi/aws.go

```diff
@@ -522,3 +532,250 @@ func removeS3Bucket(ctx context.Context, region string, bucketName string, endpo
+		Description: aws.String("Security group for Kubernetes ELB (openshift-ingress/router-default)"),
+		TagSpecifications: []ec2types.TagSpecification{
+			{
+				ResourceType: ec2types.ResourceTypeSecurityGroup,
+				Tags:         []ec2types.Tag{{Key: clusterTagKey, Value: clusterTagValue}},
+			},
+		},
+	}
+
+	sgOut, err := ec2Client.CreateSecurityGroup(ctx, sgInput)
+	if err != nil {
+		return fmt.Errorf("failed to create security group for ingress elb: %w", err)
+	}
+	securityGroupID := sgOut.GroupId
+	logrus.Infof("CCM: created security group for ingress LB: %s", *securityGroupID)
+
+	// CCM: Authorize access to port 80 and 443 to anywhere IPv4 and IPv6.
+	_, err = ec2Client.AuthorizeSecurityGroupIngress(ctx, &ec2.AuthorizeSecurityGroupIngressInput{
+		GroupId: securityGroupID,
+		IpPermissions: []ec2types.IpPermission{
+			{FromPort: aws.Int32(80), ToPort: aws.Int32(80), IpProtocol: aws.String("tcp"), IpRanges: []ec2types.IpRange{{CidrIp: aws.String(capiutils.AnyIPv4CidrBlock.String())}}},
+			{FromPort: aws.Int32(80), ToPort: aws.Int32(80), IpProtocol: aws.String("tcp"), Ipv6Ranges: []ec2types.Ipv6Range{{CidrIpv6: aws.String(capiutils.AnyIPv6CidrBlock.String())}}},
+			{FromPort: aws.Int32(443), ToPort: aws.Int32(443), IpProtocol: aws.String("tcp"), IpRanges: []ec2types.IpRange{{CidrIp: aws.String(capiutils.AnyIPv4CidrBlock.String())}}},
+			{FromPort: aws.Int32(443), ToPort: aws.Int32(443), IpProtocol: aws.String("tcp"), Ipv6Ranges: []ec2types.Ipv6Range{{CidrIpv6: aws.String(capiutils.AnyIPv6CidrBlock.String())}}},
+		},
+	})
+	if err != nil {
+		return fmt.Errorf("failed to authorize security group ingress rules for ingress elb: %w", err)
+	}
+	logrus.Info("CCM: added ingress rules for 80/443 TCP for ingress LB security group")
+
+	// CCM: Create the dual stack NLB for ingress
+	// FIXME: Let's ignore the IPv4 IP pool.
+	subnetIDs := infraCluster.Spec.NetworkSpec.Subnets.FilterPublic().IDs()
+	scheme := elbv2types.LoadBalancerSchemeEnumInternetFacing
+	if !in.InstallConfig.Config.PublicIngress() {
+		subnetIDs = infraCluster.Spec.NetworkSpec.Subnets.FilterPrivate().IDs()
+		scheme = elbv2types.LoadBalancerSchemeEnumInternal
+	}
+
+	lbInput := &elbv2.CreateLoadBalancerInput{
+		Name:          aws.String(fmt.Sprintf("%s-ingress-lb", in.InfraID)),
+		Type:          elbv2types.LoadBalancerTypeEnumNetwork,
+		Subnets:       subnetIDs,
+		Scheme:        scheme,
+		IpAddressType: elbv2types.IpAddressTypeDualstack,
+		Tags: []elbv2types.Tag{
+			{Key: clusterTagKey, Value: clusterTagValue},
+		},
+		SecurityGroups: []string{aws.ToString(securityGroupID)},
+	}
+
+	lbOut, err := elbv2Client.CreateLoadBalancer(ctx, lbInput)
+	if err != nil {
+		return fmt.Errorf("failed to create ingress dualstack nlb: %w", err)
+	}
+	lbSpec := lbOut.LoadBalancers[0]
+	logrus.Infof("CCM: created dualstack ingress LB: %s. Waiting to be ready...", *lbSpec.LoadBalancerArn)
+
+	// Wait till the load balancer becomes ready (5 minutes)
```

Comment:
Is this always true or should we attempt a backoff maybe a limited exponential backoff.

Reply:
Oh no, this is a "hack" to imitate what AWS CCM would do when it supports dualstack environment.

This code is meant to be removed so no worries :sweat: